### PR TITLE
feature: Environment variables for all AbstractContainers

### PR DESCRIPTION
**Pull request by remmelt**
Pulled the bookkeeping and creation of env vars up to AbstractContainer so all containers can set env vars.

Also ignored the Consul service from being registered in Consul.

Any container that inherits from `AbstractContainer` can now add to the `protected envVars Map`. Then call `.withEnv(createEnvironment())` and you are good.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -16,9 +16,9 @@ deps:
 
 clean:
 	./gradlew clean
-	docker rmi containersol/minimesos:latest
-	docker rmi containersol/mesos-hello-world-scheduler:latest
-	docker rmi containersol/mesos-hello-world-executor:latest
+	-docker rmi containersol/minimesos:latest
+	-docker rmi containersol/mesos-hello-world-scheduler:latest
+	-docker rmi containersol/mesos-hello-world-executor:latest
```
Comment (sadovnikov) [inline Makefile]: What do these "-" change?

File: minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java [hunk 2]
```diff
@@ -228,6 +232,10 @@ public String getClusterId() {
         return (cluster != null) ? cluster.getClusterId() : null;
     }
 
+    protected String[] createEnvironment() {
+        return envVars.entrySet().stream().map(e -> e.getKey() + "=" + e.getValue()).toArray(String[]::new);
```
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java]: `e.getValue()` is not sufficient. If the value contains " ", we'll have fun here. I expect there should be common libraries for escaping values of environment variables.

**Comment by sadovnikov** [comment]
Perfect idea, **thanks**

File: minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java (hunk 2 above)
Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java]: For one of my previous projects I wrote this
```
    private static final String SINGLE_QUOTE = "\'";
    private static final String DOUBLE_QUOTE = "\"";

    private static final String ESCAPED_SINGLE_QUOTE = "'\"'\"'";
    private static final String ESCAPED_DOUBLE_QUOTE = "\\\"";

    private static final String ESCAPED_SINGLE_QUOTE_REGEX = ESCAPED_SINGLE_QUOTE;
    private static final String ESCAPED_DOUBLE_QUOTE_REGEX = "\\\\\"";

    /**
     * Put quotes around the given String if necessary.
     * <p>
     * Originally copied from {@link org.apache.commons.exec.util.StringUtils#quoteArgument(String)}, as it does not seem to handle arguments with escaped values
     * </p>
     *
     * @param argument the argument to be quoted
     * @return the quoted argument
     * @throws IllegalArgumentException If argument contains both types of quotes
     */
    public static String quoteArgument(final String argument) {

        String cleanedArgument = removeOuterQuotes( argument, true );
        String noEscapesArgument = cleanedArgument.replaceAll( ESCAPED_SINGLE_QUOTE_REGEX, "").replaceAll( ESCAPED_DOUBLE_QUOTE_REGEX, "");

        final StringBuilder buf = new StringBuilder();
        if (noEscapesArgument.contains(DOUBLE_QUOTE)) {
            if (noEscapesArgument.contains(SINGLE_QUOTE)) {
                throw new IllegalArgumentException(
                        "Can't handle single and double quotes in same argument");
            }
            return buf.append(SINGLE_QUOTE).append(cleanedArgument).append(
                    SINGLE_QUOTE).toString();
        } else if (noEscapesArgument.contains(SINGLE_QUOTE)
                || noEscapesArgument.contains(" ")) {
            return buf.append(DOUBLE_QUOTE).append(cleanedArgument).append(
                    DOUBLE_QUOTE).toString();
        } else {
            return cleanedArgument;
        }

    }

    private static String removeOuterQuotes( String argument, boolean trim ) {

        String cleanedArgument = (trim) ? argument.trim() : argument;

        if (cleanedArgument.startsWith(SINGLE_QUOTE) && cleanedArgument.endsWith(SINGLE_QUOTE)) {
            return removeOuterQuotes(cleanedArgument.substring(1, cleanedArgument.length() - 1), false);
        }

        if (cleanedArgument.startsWith(DOUBLE_QUOTE) && cleanedArgument.endsWith(DOUBLE_QUOTE)) {
            return removeOuterQuotes(cleanedArgument.substring(1, cleanedArgument.length() - 1), false);
        }

        return cleanedArgument;

    }
```
although I'm failing to find unit tests for this

File: Makefile (hunk 1 above)
Comment (remmelt) [inline Makefile]: Ignore non-zero exit codes for the command. If the image does not exist, this is fine because we wanted to delete it anyway.

File: minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java (hunk 2 above)
Comment (remmelt) [inline minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java]: Can we move this to a different PR?

Comment (sadovnikov) [inline minimesos/src/main/java/com/containersol/minimesos/container/AbstractContainer.java]: OK, created #312
